ethercat-hal/src/io/serial_interface.rs

```diff
@@ -0,0 +1,93 @@
+use anyhow::Error;
+use smol::lock::RwLock;
+use std::{fmt, future::Future, pin::Pin, sync::Arc};
+
+pub struct SerialInterface {
+    pub has_message: Box<dyn Fn() -> Pin<Box<dyn Future<Output = bool> + Send>> + Send + Sync>,
```

Comment:
Can we remove `has_message` or do we need it since `read_message` is an `Option<Vec<u8>>`? Does `read_message` pop a queue and `has_message` does not?